src/sentry/options/manager.py

```diff
@@ -81,53 +105,78 @@ def set(self, key, value):
         )
 
         try:
-            self.update_cached_value(key, value)
+            self.update_cached_value(opt.cache_key, value)
+            return True
         except Exception:
             self.logger.warn(CACHE_UPDATE_ERR, key, exc_info=True)
+            return False
+
+    def lookup_key(self, key):
+        try:
+            return self.registry[key]
+        except KeyError:
+            # HACK: Historically, Options were used for random adhoc things.
+            # Fortunately, they all share the same prefix, 'sentry:', so
+            # we special case them here and construct a faux key until we migrate.
+            if key[:7] == 'sentry:':
+                self.logger.info('Using legacy key: %s', key, exc_info=True)
+                # History shows, there was an expectation of no types, and empty string
+                # as the default response value
+                return Key(key, '', object, self.FLAG_DEFAULT, self._make_cache_key(key))
+            raise UnknownOption(key)
 
     def get(self, key):
         """
         Get the value of an option prioritizing the cache, then the database,
         and finally the local configuration.
 
-        If no value is present for the key, an empty value ('') is returned.
+        If no value is present for the key, the default Option value is returned.
 
         >>> from sentry import options
         >>> options.get('option')
         """
-        cache_key = self._make_cache_key(key)
+        opt = self.lookup_key(key)
+
+        if not (opt.flags & self.FLAG_NOSTORE):
+            result = self.fetch_from_store(opt)
+            if result is not None:
+                return result
 
+        # Some values we don't want to allow them to be configured through
+        # config files and should only exist in the datastore
+        if opt.flags & self.FLAG_STOREONLY:
+            return opt.default
+
+        try:
+            # default to the hardcoded local configuration for this key
+            return settings.SENTRY_OPTIONS[key]
+        except KeyError:
+            return opt.default
+
+    def fetch_from_store(self, opt):
         try:
-            result = self.cache.get(cache_key)
+            result = self.cache.get(opt.cache_key)
         except Exception:
-            self.logger.warn(CACHE_FETCH_ERR, key, exc_info=True)
+            self.logger.warn(CACHE_FETCH_ERR, opt.name, exc_info=True)
             result = None
-            cache_success = False
-        else:
-            cache_success = True
 
         if result is None:
```

Comment:
This is bordering on pedantic, but technically you could declare `type=(type(None), basestring))` which would allow it to act as a union and the type declaration wouldn't error.

(Aside, TIL that `NoneType` is not available without importing from `types`.)